src/TrackGenerator.cpp

```diff
@@ -1047,6 +1049,58 @@ void TrackGenerator::initializeTrackUids() {
 
 
 /**
+ * @brief Computes the volumes/areas of each Cell and Material in the Geometry.
+ * @details This computes the volumes/areas of each Cell and Material in the 
+ *          Geometry from the length of track segments crossing each Cell. This
+ *          routine assigns the volume and number of instances to each Cell and
+ *          Material. This is a helper routine that is called after track 
+ *          segmentation is complete in TrackGenerator::generateTracks().
+ */
+void TrackGenerator::initializeVolumes() {
+
+  if (!containsTracks())
+    log_printf(ERROR, "Unable to initialize volumes since tracks "
+               "have not yet been generated");
+
+  int num_FSRs = _geometry->getNumFSRs();
+  Cell* cell;
+  Material* material;
+
+  /* Compute volume and number of instances for each Cell and Material */
+  for (int i=0; i < num_FSRs; i++) {
+    cell = _geometry->findCellContainingFSR(i);
+    cell->setVolume(cell->getVolume() + getFSRVolume(i));
```

Comment:
This is going to be very slow since it has to look over all the segments for every fsr. How about caching the fsr volumes computed by integrating the volume of the segments?